WordPressAuthenticator/Authenticator/WordPressAuthenticatorDelegateProtocol.swift

```diff
@@ -90,6 +90,11 @@ public protocol WordPressAuthenticatorDelegate: AnyObject {
     ///
     func sync(credentials: AuthenticatorCredentials, onCompletion: @escaping () -> Void)
 
+    /// Signals to the Host App that a WordPress site is available and needs validated.
+    /// This method is only triggered in the site discovery flow.
+    ///
+    func troubleshootSite(_ siteInfo: WordPressComSiteInfo?, in navigationController: UINavigationController?)
```

Comment:
Nit:

Kindly consider adding a comment about when `siteInfo` will be `nil`.

OR

Instead of having `siteInfo: WordPressComSiteInfo?` as an optional, what do you think about having another method for handling the case of no site info available scenario? i.e. `troubleshootSite` will act as success flow and the new method will act as failure flow of site discovery process.

Reply:
Updated the comment in 032faba2260b95334d96b3b9597901075fc6a0fe. I'd prefer keeping the optional site info to make it simple to work with the delegate, and also match what we have in `shouldPresentUsernamePasswordController`.